### PR TITLE
[RIPD-804] Remove dead generator map code

### PR DESCRIPTION
**Pull request by josh-ripple**
https://ripplelabs.atlassian.net/browse/RIPD-804

* Axe a bunch of unused code
* Incorporate `setGenerator()` into `createGeneratorPublic()` (`RippleAddress`)
* Cull unused args to `accountFromString()` (ledger and netOps)

### CODE REVIEW

File: src/ripple/rpc/impl/AccountFromString.cpp [hunk 1]
```diff
@@ -29,13 +29,11 @@ namespace RPC {
 //
 // Returns a Json::objectValue, containing error information if there was one.
 Json::Value accountFromString (
-    Ledger::ref lrLedger,
     RippleAddress& naAccount,
     bool& bIndex,
```
Comment (nbougalis) [inline src/ripple/rpc/impl/AccountFromString.cpp]: I believe that it is safe to also remove `bIndex` and `iIndex` (use `0`) from the argument list.

**Comment by nbougalis** [comment]
Needs a good commit message when squashed together. Other than that :+1:

**Comment by josh-ripple** [comment]
User-visible changes:  `account_index` and `peer_index` (marked deprecated) are removed.  (They'll be ignored with no error reported.)  `accountFromString()` no longer accepts a seed as a substitute for a public identifier.

**Comment by nbougalis** [comment]
This needs one more reviewer.

**Comment by rec** [comment]
I'll do that!

File: src/ripple/protocol/impl/RippleAddress.cpp [hunk 2]
```diff
@@ -706,16 +706,13 @@ std::string RippleAddress::humanGenerator () const
...
-    mIsValid        = true;
-    SetData (VER_FAMILY_GENERATOR, vPublic);
-}
-
 RippleAddress RippleAddress::createGeneratorPublic (RippleAddress const& naSeed)
 {
     RippleAddress   naNew;
-    naNew.setGenerator (GenerateRootDeterministicPublicKey (naSeed.getSeed()));
+    
+    auto publicKey = GenerateRootDeterministicPublicKey (naSeed.getSeed());
```
Comment (rec) [inline src/ripple/protocol/impl/RippleAddress.cpp]: If `GenerateRootDeterministicPublicKey` is a function, it should start with a lower case letter...  not part of this change but just noted.

Comment (rec) [inline src/ripple/protocol/impl/RippleAddress.cpp]: Sent out a quick cleanup for that.  :-D

File: src/ripple/rpc/handlers/AccountLines.cpp [hunk 3]
```diff
@@ -86,12 +85,10 @@ Json::Value doAccountLines (RPC::Context& context)
         return result;
 
     std::string strIdent (params[jss::account].asString ());
-    bool bIndex (params.isMember (jss::account_index));
-    int iIndex (bIndex ? params[jss::account_index].asUInt () : 0);
     RippleAddress rippleAddress;
 
-    Json::Value const jv (RPC::accountFromString (ledger, rippleAddress, bIndex,
-        strIdent, iIndex, false, context.netOps));
+    Json::Value const jv (RPC::accountFromString (rippleAddress,
```
Comment (rec) [inline src/ripple/rpc/handlers/AccountLines.cpp]: why not put all the parameters on the same line together?

why not use auto?  you could probably fit it all on one line...!

File: src/ripple/rpc/handlers/AccountLines.cpp [hunk 4]
```diff
@@ -114,11 +109,8 @@ Json::Value doAccountLines (RPC::Context& context)
     {
         result[jss::peer] = rippleAddress.humanAccountID ();
 
-        if (bPeerIndex)
-            result[jss::peer_index] = iPeerIndex;
-
-        result = RPC::accountFromString (ledger, rippleAddressPeer, bPeerIndex,
-            strPeer, iPeerIndex, false, context.netOps);
+        result = RPC::accountFromString (rippleAddressPeer,
```
Comment (rec) [inline src/ripple/rpc/handlers/AccountLines.cpp]: This fits fine on one line...

File: src/ripple/rpc/handlers/AccountLines.cpp (hunk 3 above)
Comment (rec) [inline src/ripple/rpc/handlers/AccountLines.cpp]: I have the same comment multiple times below.

**Comment by rec** [comment]
:+1:  You could contemplate the suggested changes and do them or not.

**Comment by rec** [comment]
If you rebased this, it could go into the branch I'm about to prepare.

File: src/ripple/rpc/handlers/AccountLines.cpp (hunk 3 above)
Comment (josh-ripple) [inline src/ripple/rpc/handlers/AccountLines.cpp]: The ones named `jv` I tagged `auto`.  The ones named `jvAccepted` I renamed to `error`.  In both cases, that made the unbroken lines short enough.  Calls with each argument on its own line I left alone.

**Comment by josh-ripple** [comment]
Obviously, "Replace AccountState's addJson() with getJson()" has to happen before "Rename jvAccepted to error", and it does in the Git repository.  GitHub has them backwards.

**Comment by rec** [comment]
Github sometimes lists the orders in reverse here...

**Comment by rec** [comment]
:+1:

**Comment by nbougalis** [comment]
Looks good to me :+1:

**Comment by rec** [comment]
THink the failure is spurious, so I kicked travis...

**Comment by nbougalis** [comment]
Personally, I'd squash this down into a single commit, or at most two (with the top four or five together into one).

**Comment by josh-ripple** [comment]
I squashed the generator map culling and related cleanup to one commit.  The remainder is out of scope for RIPD-804.

**Comment by rec** [comment]
LGTM!  :+1:

**Comment by nbougalis** [comment]
:+1:

**Comment by nbougalis** [comment]
Merged as daf4f8fcdeea48bed08c25a7586f767c056450b4
